RFCs/rfc-0004-fast-transactions.md

```diff
@@ -0,0 +1,77 @@
+# RFC-0004: Decreasing the minimum interval between SmartContract's Transactions from 3 minutes to 1 second with Recursive Proofs
+
+- **Intent**: Implement recursive proofs in SmartContract transactions to reduce the minimum transaction interval from 3 minutes to 1 second.
+- **Submitted by**: DFST ([Github](https://github.com/dfstio), [Twitter](https://twitter.com/dfst_io), Discord: dfst.io)
+- **Date**: Thursday, December 07, 2023
+
+## Abstract
+
+This RFC aims to improve the Mina Protocol's developer and user experience by enabling faster transaction processing in SmartContracts. Currently, a minimum interval of 3 minutes is required between SmartContract transactions that have preconditions and alter the state. This proposal intends to shorten this interval to 1 second, allowing rapid transaction execution with state preconditioning and changing.
+
+## Introduction
+
+Mina Protocol's SmartContract transactions, authenticated through proof, must meet these criteria:
+
+- Matching account update preconditions with the current account state.
+- Proof verification against the SmartContract's verification key.
+- Fee alignment with the market rate.
+
+Once verified, the transaction's inclusion in the block is assured, bypassing the need for blockchain-wide recalculations. Despite a known post-verification state, SmartContracts must wait about 3 minutes for state reflection in the new block before initiating another transaction.
```

Comment:
Agree; I've amended the RFC.

If we are speaking about one-actor transactions (that also include the case when zkApp is the sequencer for user requests), we should also note two limitations:
- The total limit of zkApp transactions in one block is around 120.
- The actor cannot simply increase the nonce and send the new transaction to the same block in case the transaction includes preconditions checks and changes the state.

This RFC aims to shift those two limitations.